core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -235,14 +242,40 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws LimitExceededExceptio
                                                  .mapToInt(machineCfg -> machineCfg.getLimits().getRam())
                                                  .sum();
             if (workspaceRam > maxRamPerEnv) {
-                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%dmb' and you requested '%dmb'. " +
+                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%s' and you requested '%s'. " +
                                                         "This value is set by your admin with the 'limits.workspace.env.ram' property",
-                                                        maxRamPerEnv,
-                                                        workspaceRam));
+                                                        parseMegabyteSizeToString(maxRamPerEnv, "mb"),
+                                                        parseMegabyteSizeToString(workspaceRam, "mb")));
             }
         }
     }
 
+    private String parseMegabyteSizeToString(long ramSizeMB, String suffix) {
```

Comment:
Well, about the method: i don't think this is a good place for such kind of transformations. 

1. Basically you need this `parse` method  only in the one place `checkRamAndPropagateStart`
2. You don't need to parse from megabytes to megabytes.
3. If [Size]( https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/Size.java) class doesn't work appropriate for you, then you should either simplify this method to convert megabytes to gigabytes(and that's it), or add a new method into the `Size` class, which may be reused by others.